pocs/focuser/focuser.py

```diff
@@ -294,35 +293,16 @@ def autofocus(self,
             'merit_function': merit_function,
             'merit_function_kwargs': merit_function_kwargs,
             'mask_dilations': mask_dilations,
+            'coarse': coarse,
             'plots': plots,
-            'start_event': None,
-            'finished_event': None,
+            'focus_event': focus_event,
         }
-
-        # Coarse focus
-        if coarse:
-            coarse_event = Event()
-            focus_params['finished_event'] = coarse_event
-            focus_params['coarse'] = True
-
-            coarse_thread = Thread(target=self._autofocus, kwargs=focus_params)
-            coarse_thread.start()
-        else:
-            coarse_event = None
-
-        # Fine Focus - This will wait for the coarse_event to finish.
-        fine_event = Event()
-        focus_params['start_event'] = coarse_event
-        focus_params['finished_event'] = fine_event
-        focus_params['coarse'] = False
-
-        fine_thread = Thread(target=self._autofocus, kwargs=focus_params)
-        fine_thread.start()
-
+        focus_thread = Thread(target=self._autofocus, kwargs=focus_params)
+        focus_thread.start()
         if blocking:
-            fine_event.wait()
+            focus_event.wait()
```

Comment:
Should include some ultimate timeout in the `wait`

Reply:
Yes. Needs to be a rather conservative one though, as it's a bit involved to predict exactly how long it will take.

Reply:
That might be a job for another PR actually. Want to add timeouts to the camera's threaded operations more generally to prevent the lock ups we sometimes get.